Make the tests pass for v0.4 and add enum support

poros opened the pull request:
```
    Make the tests pass for v0.4 and add enum support
```
---
```
    Copy setup from avro-rs
```
---
```
    Run clippy as pre-commit hook
```
---
```
   Implement all clippy suggestion except from casting errors
```
---
```
    Allow cast_ptr_alignment error for clippy

    Clippy is complaining a lot about many of the casts the library is
    doing. This happens because when we define something like `pub struct
    AvroWriter` to be the opaque type in the C bindings and then we cast
    `avro_rs::Writer` from/into it, the compiler has no idea that these two
    pointers actually point to structs of the same type.

    Unfortunately we cannot use `pub type AvroWriter = Writer` because that
    would be translated by cbindgen in `typedef Writer AvroWriter`, with
    Writer being a missing symbol (since it is only defined in the
    dependency). Instead what we want is `typedef struct AvroWriter
    AvroWriter`, which is exactly what `pub struct AvroWriter` is giving us.
```
---
```
    Try to persuade travis to install cbindgen
```

poros committed:
Make the tests pass for v0.4 and add enum support

flavray commented on src/avro_utils.rs:
```diff
@@ -196,7 +217,10 @@ pub fn pickle_value_from_avro(value: Value) -> PickleValue {
                 .collect::<BTreeMap<_, _>>(),
         ),
-        Value::Enum(_) => PickleValue::None, // not yet supported
+        Value::Enum(idx, repr) => PickleValue::Tuple(vec![
```
It seems a bit odd to return a `(index, symbol)` tuple when encoding while we expect a `symbol` string when decoding.

I see that [`avro`](https://github.com/Yelp/yelp-avro/blob/trunk/lang/py/src/avro/io.py) return only the `symbol`. Would it make sense to do the same?

poros replied:
Oh, yes, absolutely. That was just an almost random decision

poros committed:
Copy setup from avro-rs

poros committed:
Run clippy as pre-commit hook

poros committed:
Implement all clippy suggestion except from casting errors

poros committed:
Allow cast_ptr_alignment error for clippy

Clippy is complaining a lot about many of the casts the library is
doing. This happens because when we define something like `pub struct
AvroWriter` to be the opaque type in the C bindings and then we cast
`avro_rs::Writer` from/into it, the compiler has no idea that these two
pointers actually point to structs of the same type.

Unfortunately we cannot use `pub type AvroWriter = Writer` because that
would be translated by cbindgen in `typedef Writer AvroWriter`, with
Writer being a missing symbol (since it is only defined in the
dependency). Instead what we want is `typedef struct AvroWriter
AvroWriter`, which is exactly what `pub struct AvroWriter` is giving us.

poros commented:
Tests are segfaulting on Travis but are passing on my laptop :/
Would you mind trying it out yourself? I haven't much clue of how to debug such a thing.

poros committed:
Install bindgen in travis

poros committed:
Madness

poros committed:
Madness intensifies
